fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAnalyzerRegistryImpl.java

```diff
@@ -86,17 +88,41 @@ private LuceneAnalyzerRegistryImpl() {
 
     @Nonnull
     @Override
-    public Pair<AnalyzerChooser, AnalyzerChooser> getLuceneAnalyzerChooserPair(@Nonnull Index index, @Nonnull LuceneAnalyzerType type) {
-        final String name = index.getOption(type.getIndexOptionKey());
+    public LuceneAnalyzerCombinationProvider getLuceneAnalyzerCombinationProvider(@Nonnull Index index, @Nonnull LuceneAnalyzerType type) {
+        final String defaultAnalyzerName = index.getOption(type.getAnalyzerOptionKey());
+        final String analyzerPerFieldName = index.getOption(type.getAnalyzerPerFieldOptionKey());
+        Pair<AnalyzerChooser, AnalyzerChooser> defaultAnalyzerChooserPair = getAnalyzerChooser(index, defaultAnalyzerName, type);
+
+        if (analyzerPerFieldName == null) {
+            return new LuceneAnalyzerCombinationProvider(defaultAnalyzerChooserPair.getLeft(), defaultAnalyzerChooserPair.getRight(),
+                    null, null);
+        }
+
+        Map<String, AnalyzerChooser> indexAnalyzerChooserPerFieldOverride = new TreeMap<>();
+        Map<String, AnalyzerChooser> queryAnalyzerChooserPerFieldOverride = new TreeMap<>();
+        String[] elements = analyzerPerFieldName.split("/");
```

Comment:
@alecgrieser yeah it sounds better to use different delineators for that between key-value pairs and that between keys and values. So can I assume no field names will have ```:``` or ```,```?

Reply:
I don't know how valid an assumption that is; I think `:` and `,` are better delineator characters for maps, but I don't know whether they're currently allowed in field names (or analyzer names) or not. That's one of the reasons that for, say, the "replaced by" index name options, rather than coming up with a scheme by which the index names could be put together in a single option, we went with instead a single option _prefix_, and then collected all IndexOptions that began with that prefix to produce the list.